crucible-llvm/src/Lang/Crucible/LLVM/Utils.hs

```diff
@@ -0,0 +1,15 @@
+------------------------------------------------------------------------
+-- |
+-- Module           : Lang.Crucible.LLVM.Utils
+-- Description      : Miscellaneous utility functions.
+-- Copyright        : (c) Galois, Inc 2021
+-- License          : BSD3
+-- Maintainer       : Rob Dockins <rdockins@galois.com>
+-- Stability        : provisional
+------------------------------------------------------------------------
+module Lang.Crucible.LLVM.Utils (applyUnless) where
+
+-- | If the first argument is 'False', apply the second argument to the third.
+-- Otherwise, simply return the third argument.
+applyUnless :: Applicative f => Bool -> (a -> f a) -> a -> f a
```

Comment:
I wasn't sure of a good place to put this, so I invented a new module for my needs. Let me know if there is a better home for this.

Reply:
*shrug* Seems OK to me.  I haven't seen this exact function before, so I don't know if there's some better place to grab it from.